_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp

```diff
@@ -653,11 +653,9 @@ namespace ExtBuffer
 
             return false;
         }
-        if (buf.PicWidthInLumaSamples == 0)
-            buf.PicWidthInLumaSamples  = align2_value(par.mfx.FrameInfo.CropW > 0 ? (mfxU16)(par.mfx.FrameInfo.CropW + par.mfx.FrameInfo.CropX)  : par.mfx.FrameInfo.Width,  codedPicAlignment);
 
-        if (buf.PicHeightInLumaSamples == 0)
-            buf.PicHeightInLumaSamples = align2_value(par.mfx.FrameInfo.CropH > 0 ? (mfxU16)(par.mfx.FrameInfo.CropH + par.mfx.FrameInfo.CropY)  : par.mfx.FrameInfo.Height, codedPicAlignment);
+        buf.PicWidthInLumaSamples  = align2_value(par.mfx.FrameInfo.CropW > 0 ? (mfxU16)(par.mfx.FrameInfo.CropW + par.mfx.FrameInfo.CropX)  : par.mfx.FrameInfo.Width,  codedPicAlignment);
```

Comment:
Are sure that this is correct fix? In my understanding you will overwrite data provided by user in buffer ```mfxExtHEVCParam```, dig into line 649 for details

Reply:
Yes, indeed. I fixed this, now PicHeightInLumaSamples and PicWidthInLumaSamples will be updated only when they contain incorrect data.